jetty-core/jetty-util/src/test/java/org/eclipse/jetty/util/resource/JarResourceTest.java

```diff
@@ -72,8 +73,8 @@ public void testJarFile()
         {
             Resource r = resourceFactory.newResource(uri);
 
-            Set<String> entries = new HashSet<>(r.list());
-            assertThat(entries, containsInAnyOrder("alphabet", "numbers", "subsubdir/"));
+            Set<String> entries = Files.list(r.getPath()).map(Path::getFileName).map(Path::toString).collect(Collectors.toSet());
+            assertThat(entries, containsInAnyOrder("alphabet", "numbers", "subsubdir"));
```

Comment:
IIRC the trailing slash is needed when directory listing is served by the `DefaultServlet`.

Reply:
I bet we have test cases for that.
I'll check.

Reply:
Yep, we have testcases for `DefaultServlet`.

In EE10, it uses the `ResourceService`, which in turn uses the `ResourceListing` class, which manages the trailing slash itself.
I'll check EE9/EE8 next.

Reply:
The EE9/EE8 `DefaultServlet` uses the `ResourceListing` class from `jetty-core/jetty-server` as well.
No trailing slash problem there too.